compiler/luci/pass/src/ForwardReshapeToUnaryOpPass.cpp

```diff
@@ -180,6 +267,90 @@ class ForwardReshape final : public luci::CircleNodeMutableVisitor<bool>
 
     return forward_reshape(reshape, node);
   }
+
+  bool visit(luci::CircleDiv *node)
+  {
+    luci::CircleReshape *reshape = nullptr;
+    luci::CircleConst *const_value = nullptr;
+
+    if (not luci::fill(&reshape, &const_value).with_commutative_args_of(node))
+      return false;
+
+    if (const_value->dtype() != loco::DataType::FLOAT32)
+      return false;
+
+    // Should be scalar
+    if (const_value->size<loco::DataType::FLOAT32>() != 1)
+      return false;
+
+    return forward_reshape(reshape, node, const_value);
+  }
+
+  bool visit(luci::CircleMul *node)
+  {
+    luci::CircleReshape *reshape = nullptr;
+    luci::CircleConst *const_value = nullptr;
+
+    if (not luci::fill(&reshape, &const_value).with_commutative_args_of(node))
+      return false;
+
+    if (const_value->dtype() != loco::DataType::FLOAT32)
+      return false;
+
+    // Should be scalar
+    if (const_value->size<loco::DataType::FLOAT32>() != 1)
+      return false;
+
+    return forward_reshape(reshape, node, const_value);
+  }
+
+  bool visit(luci::CircleMean *node)
+  {
+    luci::CircleReshape *reshape = nullptr;
+    luci::CircleConst *axis = nullptr;
+
+    reshape = dynamic_cast<luci::CircleReshape *>(node->input());
+    axis = dynamic_cast<luci::CircleConst *>(node->reduction_indices());
+
+    if (reshape == nullptr or axis == nullptr)
+      return false;
+
+    if (axis->dtype() != loco::DataType::S32)
+      return false;
+
+    // Should be scalar
+    if (axis->size<loco::DataType::S32>() != 1)
+      return false;
+
+    // axis value
+    int32_t axis_value;
+    axis_value = axis->at<loco::DataType::S32>(0);
+
+    if (axis_value < 0)
+      axis_value += static_cast<int32_t>(reshape->rank());
+
+    assert(axis_value >= 0);
+
+    if (node->keep_dims() != true)
+      return false;
+
+    auto *reshape_input = dynamic_cast<luci::CircleNode *>(reshape->tensor());
+    if (reshape_input == nullptr)
+      return false;
+
+    // reshape shouldn't change rank
+    if (reshape_input->rank() != reshape->rank())
+      return false;
+
+    assert(reshape_input->rank() > static_cast<uint32_t>(axis_value));
+
+    // axis value for mean should be the same after reshape
+    if (reshape_input->dim(axis_value).value() != reshape->dim(axis_value).value())
+      return false;
```

Comment:
Reshape can change the start address of each dimension., so this condition is not enough.

```python
import numpy as np

# input
a = np.random.rand(2,2,3)

# Before
b = np.reshape(a, (3,2,2))
c = np.mean(b, 1, keepdims=True)

# After
d = np.mean(a, 1, keepdims=True)
e = np.reshape(d, (3,1,2))

print(np.allclose(c, e, rtol=1e-5, atol=1e-5))                                        
```

Return value is false.

 For simplicity, you can check the axis and inner dimensions (0~axis_value) of the input/output of Reshape Op are the same.